### PR TITLE
2195 cargos

### PR DESCRIPTION
**Pull request by levBagryansky**
Closes: #2195 
Create separate cargo project for every rust insert.

---

### Focus of this PR:
This PR focuses on improving the Rust integration in the codebase.

### Detailed summary:
- Updated the `rand` dependency from version `0.5.5` to `0.5.2` in `eo-maven-plugin/src/test/resources/org/eolang/maven/twice-rust.eo`.
- Added the `with` method in `eo-maven-plugin/src/main/java/org/eolang/maven/rust/Project.java` to return the project object.
- Updated the `EOrust` class in `eo-runtime/src/main/java/EOorg/EOeolang/EOrust.java` to load Rust libraries from multiple directories.
- Modified the `exec` method in `eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java` to create a directory for each Rust project and save the project.
- Updated the `exec` method in `eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java` to build each Rust project in parallel.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by levBagryansky** [comment]
@Graur please review. Sems there were troubles with connection in windows test: 
```
Error: [ERROR] There was an error in compilation
Warning:  >>     Updating crates.io index
Warning: [WARNING] >> warning: spurious network error (3 tries remaining): [28] Timeout was reached (Operation too slow. Less than 10 bytes/sec transferred the last 30 seconds)
Warning: [WARNING] >> warning: spurious network error (2 tries remaining): [28] Timeout was reached (Operation too slow. Less than 10 bytes/sec transferred the last 30 seconds)
Warning: [WARNING] >> warning: spurious network error (1 tries remaining): [28] Timeout was reached (Operation too slow. Less than 10 bytes/sec transferred the last 30 seconds)
Warning:  >> error: failed to get `syn` as a dependency of package `thiserror-impl v1.0.40`
Warning:  >>     ... which satisfies dependency `thiserror-impl = "=1.0.40"` of package `thiserror v1.0.40`
Warning:  >>     ... which satisfies dependency `thiserror = "^1.0.20"` of package `jni v0.21.1`
Warning:  >>     ... which satisfies dependency `jni = "^0.21.1"` of package `common v0.1.0 (D:\a\eo\eo\eo-runtime\target\eo-test\Lib\dxxaxeoxeoxeoxruntimextargetxeoxtest0)`
Warning:  >> 
Warning:  >> Caused by:
Warning:  >>   failed to query replaced source registry `crates-io`
Warning:  >> 
Warning:  >> Caused by:
Warning:  >>   download of 3/s/syn failed
Warning:  >> 
Warning:  >> Caused by:
Warning:  >>   failed to download from `[https://index.crates.io/3/s/syn`](https://index.crates.io/3/s/syn%60)
Warning:  >> 
Warning:  >> Caused by:
Warning:  >>   [28] Timeout was reached (Operation too slow. Less than 10 bytes/sec transferred the last 30 seconds)
Error:  Failed to build cargo project with dest = D:\a\eo\eo\eo-runtime\target\eo-test\Lib\dxxaxeoxeoxeoxruntimextargetxeoxtest0
```

**Comment by levBagryansky** [comment]
@Graur please take a look

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 1]
```diff
@@ -75,42 +78,45 @@ public final class BinarizeMojo extends SafeMojo {
...
+                        String.format(
+                            "Interrupted while building %s",
+                            file
+                        ),
+                        exception
+                    );
+                }
+                if (building.exitValue() != 0) {
+                    Logger.error(this, "There was an error in compilation");
+                    final ByteArrayOutputStream stdout = new ByteArrayOutputStream();
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:101]: @levBagryansky Same here. Looks like we can inline this variable. According to: https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html

**Review by Graur — CHANGES_REQUESTED** [review]
@levBagryansky Good one! Just a few comments from my side:

**Review by Graur — APPROVED** [review]
@levBagryansky Looks good to me
@yegor256 Could you please have a look?

**Comment by levBagryansky** [comment]
@yegor256 please take a look

**Comment by levBagryansky** [comment]
@yegor256 please check

**Comment by levBagryansky** [comment]
@yegor256 could you please merge these changes?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**